circuits/cpp/src/aztec3/constants.hpp

```diff
@@ -185,4 +185,79 @@ enum PrivateStateNoteGeneratorIndex {
+// https://github.com/noir-lang/noir/issues/1734
+constexpr size_t L1_TO_L2_MESSAGE_LENGTH = 8;
+// message length + sibling path (same size as tree height) + 1 field for root + 1 field for index
+constexpr size_t L1_TO_L2_MESSAGE_ORACLE_CALL_LENGTH = L1_TO_L2_MESSAGE_LENGTH + L1_TO_L2_MSG_TREE_HEIGHT + 1 + 1;
+
+// TODO: Remove these when nested array is supported.
+constexpr size_t MAX_NOTE_FIELDS_LENGTH = 20;
+// + 2 for EXTRA_DATA: [number_of_return_notes, contract_address]
+constexpr size_t GET_NOTE_ORACLE_RETURN_LENGTH = MAX_READ_REQUESTS_PER_CALL * MAX_NOTE_FIELDS_LENGTH + 2;
+constexpr size_t MAX_NOTES_PER_PAGE = 10;
+// + 2 for EXTRA_DATA: [number_of_return_notes, contract_address]
+constexpr size_t VIEW_NOTE_ORACLE_RETURN_LENGTH = MAX_NOTES_PER_PAGE * MAX_NOTE_FIELDS_LENGTH + 2;
+
+constexpr size_t CALL_CONTEXT_LENGTH = 6;
+constexpr size_t COMMITMENT_TREES_ROOTS_LENGTH = 4;
+constexpr size_t FUNCTION_DATA_LENGTH = 3;
+constexpr size_t CONTRACT_DEPLOYMENT_DATA_LENGTH = 4;
+
+// PRIVATE_CIRCUIT_PUBLIC_INPUTS_LENGTH
+// CALL_CONTEXT_LENGTH - 6
+// + ARGS_HASH - 1
+// + RETURN_VALUES_LENGTH - 4
+// + MAX_READ_REQUESTS_PER_CALL - 4
+// + MAX_NEW_COMMITMENTS_PER_CALL - 4
+// + MAX_NEW_NULLIFIERS_PER_CALL - 4
+// + MAX_PRIVATE_CALL_STACK_LENGTH_PER_CALL - 4
+// + MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL - 4
+// + MAX_NEW_L2_TO_L1_MSGS_PER_CALL - 2
+// + encrypted_logs_hash - 2
+// + unencrypted_logs_hash - 2
+// + encrypted_log_preimages_length - 1
+// + unencrypted_log_preimages_length - 1
+// + commitment_trees_roots - 4
+// + contract_deployment_data - 4
+// + chain_id - 1
+// + version - 1
+// = 6 + 1 + 4 + 4 + 4 + 4 + 4 + 4 + 2 + 2 + 2 + 1 + 1 + 4 + 4 + 1 + 1
+// = 49
+// Change this ONLY if you have changed the PrivateCircuitPublicInputs structure in C++.
+// In other words, if the structure/size of the public inputs of a function call changes then we
+// should change this constant as well as the offsets in private_call_stack_item.nr
+constexpr size_t PRIVATE_CIRCUIT_PUBLIC_INPUTS_LENGTH =
+    CALL_CONTEXT_LENGTH + 1 + RETURN_VALUES_LENGTH + MAX_READ_REQUESTS_PER_CALL + MAX_NEW_COMMITMENTS_PER_CALL +
+    MAX_NEW_NULLIFIERS_PER_CALL + MAX_PRIVATE_CALL_STACK_LENGTH_PER_CALL + MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL +
+    MAX_NEW_L2_TO_L1_MSGS_PER_CALL + NUM_FIELDS_PER_SHA256 + NUM_FIELDS_PER_SHA256 + 1 + 1 +
+    COMMITMENT_TREES_ROOTS_LENGTH + CONTRACT_DEPLOYMENT_DATA_LENGTH + 1 + 1;
+
+
+constexpr size_t CONTRACT_STORAGE_UPDATE_REQUEST_LENGTH = 3;
+constexpr size_t CONTRACT_STORAGE_READ_LENGTH = 2;
+
+// CALL_CONTEXT_LENGTH
+// + ARGS_HASH
+// + RETURN_VALUES_LENGTH
+// + MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_CALL * CONTRACT_STORAGE_UPDATE_REQUEST_LENGTH
+// + MAX_PUBLIC_DATA_READS_PER_CALL * CONTRACT_STORAGE_READ_LENGTH
+// + MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL
+// + MAX_NEW_COMMITMENTS_PER_CALL
+// + MAX_NEW_NULLIFIERS_PER_CALL
+// + MAX_NEW_L2_TO_L1_MSGS_PER_CALL
+// + COMMITMENT_TREES_ROOTS_LENGTH - 4
+// + 2;
```

Comment:
Do we want all the numbers here? it seems like it would be hard to figure out which to update if you change a constant. I'm good without this comment overall

Reply:
Addressed in [971eeb2](https://github.com/AztecProtocol/aztec-packages/pull/1072/commits/971eeb21465a7b197cec9d735813766df9b9c8b9)